Android build error - 'if' must have both main and 'else' branches if used as an expression

<!--
  MAKE SURE TO READ AND FOLLOW THIS TEMPLATE CLOSELY OR YOUR ISSUE WILL BE CLOSED WITHOUT NOTICE
-->

### Is this a bug report, a feature request, or a question?

Bug report

<!--
  If you answered "Bug report":

    We expect you to produce a high-quality bug report since putting care into your report helps us fix the issue faster.
    For bug reports, it is REQUIRED to fill the rest of this template, or the issue will be closed.

  If you answered "Feature request" or "Question":

    Make sure to describe as precisely as possible the feature you'd like to see implemented or the question you'd like to see answered.
    When relevant, provide visual examples (screenshots, screencasts, diagrams...).
    You can ignore the next steps as long as you've made sure that your description is as clear, thorough and illustrated as possible.
-->

### Have you followed the required steps before opening a bug report?

(Check the step you've followed - put an `x` character between the square brackets (`[]`).)

- [x] I have reviewed [the documentation](https://github.com/EkoLabs/react-native-background-downloader/blob/master/README.md) in its entirety, including the dedicated documentations :books:.
- [x] I have searched for [existing issues](https://github.com/EkoLabs/react-native-background-downloader/issues) and made sure that the problem hasn't already been reported.
- [x] I am using [the latest plugin version](https://github.com/EkoLabs/react-native-background-downloader/releases).

<!--
  Please DO NOT go further if you've not followed ALL of the above steps.
  Failing to do so will result in your issue getting closed without warning.
-->

### Is the bug specific to iOS or Android? Or can it be reproduced on both platforms?

Android

### Is the bug related to the native implementation? (NSURLSession on iOS and Fetch on Android)
<!--
  Did you read the [NSURLSession documentation](https://developer.apple.com/documentation/foundation/nsurlsession) or [Fetch documentation](https://github.com/tonyofrancis/Fetch)?
  Did you try searching for your issue in the context of NSURLSeesion or Fetch?
-->

Yes

### Environment

<!--
  Provide information about your current environment. At the very least, it should include the following:

  Environment:
    React: 16.0.0-beta.5
    React native: 0.49.2
    react-native-background-downloader: 2.1.0

  Target Platform:
    Android (6.0)
    iOS (10.3)
-->

  Environment:
    React native: 0.76.9
    react-native-background-downloader: 4.3.2
    kotlinVersion (android/build.gradle): 1.9.25

### Expected Behavior

<!--
  How did you expect your project to behave?
  It’s fine if you’re not sure your understanding is correct.
  Just write down what you thought would happen.
-->

Build process compiles the kotlin and succeeds

### Actual Behavior

<!--
  Did something go wrong?
  Is something broken, or not behaving as you expected?
  Describe this section in detail, and attach logs if possible.
  Don't just say "it doesn't work"!
-->

Android build fails with:
```
'if' must have both main and 'else' branches if used as an expression
```
at lines 598 and 607 in `RNBackgroundDownloaderModuleImpl.kt`

Logs:
```
error Failed to install the app. Command failed with exit code 1: ./gradlew app:installDebug -PreactNativeDevServerPort=8081
e: file:///Users/me/dev/repro-repos/RNBGDownloaderIssueRepro/node_modules/@kesha-antonov/react-native-background-downloader/android/src/main/java/com/eko/RNBackgroundDownloaderModuleImpl.kt:598:9 'if' must have both main and 'else' branches if used as an expression
e: file:///Users/me/dev/repro-repos/RNBGDownloaderIssueRepro/node_modules/@kesha-antonov/react-native-background-downloader/android/src/main/java/com/eko/RNBackgroundDownloaderModuleImpl.kt:607:11 'if' must have both main and 'else' branches if used as an expression FAILURE: Build failed with an exception. * What went wrong:
Execution failed for task ':kesha-antonov_react-native-background-downloader:compileDebugKotlin'.
```

### Steps to Reproduce

<!--
  How would you describe your issue to someone who doesn’t know you or your project?
  Write a sequence of steps that anybody can repeat to see the issue.
  Be specific! If the bug cannot be reproduced, your issue will be closed.
-->

1. Use the RN 0.76 version of the react-native-community/cli template (which uses kotlin 1.9.25) with:
`npx @react-native-community/cli@15 init RNBGDownloaderIssueRepro --version 0.76`
2. `npm install @kesha-antonov/react-native-background-downloader`
3. `npm run android`

Minimal reproduction repo (includes a patch in the `patches` folder that resolves the build error):
https://github.com/agerbens/RNBGDownloaderIssueRepro

The patch adds `else` branches with logging to the two `if` statements in `pauseTask()`:
1. `if (config != null)` - added `else` with warning log
2. `if (paused)` - added `else` with debug log

I'm not sure if this is happening because of the older (1.9) version of kotlin or an older version of gradle, but it doesn't seem to be happening on newer versions of the react-native template which use kotlin 2+

I can make a PR with the changes in the patch if requested. Just let me know. Thank you!

<!--
  Thanks for helping us help you!
-->

--- a/android/src/main/java/com/eko/RNBackgroundDownloaderModuleImpl.kt
+++ b/android/src/main/java/com/eko/RNBackgroundDownloaderModuleImpl.kt
@@ -643,7 +643,11 @@ class RNBackgroundDownloaderModuleImpl(private val reactContext: ReactApplicatio
             configIdToDownloadId.remove(configId)
             saveDownloadIdToConfigMap()
             Log.d(NAME, "Paused DownloadManager download: $configId")
+          } else {
+            Log.d(NAME, "pauseTask: Download not paused (may already be paused): $configId")
           }
+        } else {
+          Log.w(NAME, "pauseTask: No config found for downloadId: $downloadId")
         }
       } else {
         Log.w(NAME, "pauseTask: No download found for configId: $configId")

